### ISSUE TITLE
Escape characters confuse the escaping code

### ISSUE THREAD

**Issue by philippjfr** [ioam/holoviews#473]
Just came across a bug in the dimension escaping code. When constructing an NdOverlay like this:

```python
ndoverlay = hv.NdOverlay({i: hv.Contours([np.random.rand(20,2)]) for i in range(10)},
                         kdims=[hv.Dimension('$\alpha$')])
ndoverlay
```

I get the following error:

```
  File "/Users/prudiger/topographica/external/holoviews/holoviews/plotting/plot.py", line 501, in _get_frame
    selection = self.hmap.select((HoloMap, DynamicMap), **select)
  File "/Users/prudiger/topographica/external/holoviews/holoviews/core/dimension.py", line 722, in select
    select[self.get_dimension_index(dim)] = val
  File "/Users/prudiger/topographica/external/holoviews/holoviews/core/dimension.py", line 652, in get_dimension_index
    (dim, self.__class__.__name__))
Exception: Dimension Frame not found in HoloMap.
```

This seems to be because:

```python
ndoverlay.kdims
```

```
[Dimension('$\x07lpha$')]
```

And the ``\x07`` unicode character cannot be escaped by the escaping code in:

```
File "/Users/prudiger/topographica/external/holoviews/holoviews/core/util.py", line 249, in shortened_character_name
name = unicodedata.name(c).lower()
ValueError: no such name
```

**Comment by jlstevens** [comment]
Ok, thankfully this issue was quick to identify.

The problem is that ``\a`` is an escape code, which according to [this](http://python-reference.readthedocs.org/en/latest/docs/str/escapes.html) is ``ASCII Bell (BEL)``. The solution is to use a raw string i.e:

```python
ndoverlay = hv.NdOverlay({i: hv.Contours([np.random.rand(20,2)]) for i in range(10)},
                         kdims=[hv.Dimension('$\alpha$')])
```

This is a tricky gotcha right now. I suppose I can try to detect escape codes, apparently I can do this by checking if ``unicodedata.type`` is ``"Cn"``. After that, it would need some sort of special handling to undo the escape codes (but not all of them).

So in short, I don't think this is a bug as such but maybe a feature request. I'll have a quick look to see what I might be able to do...

**Comment by jlstevens** [comment]
I'll note that ``'$\\alpha$'`` does work so maybe that is what I should aim to do if I detect an escape code.

**Comment by jlstevens** [comment]
Ok, I have a prototype that fixes this issue in HoloViews...but that doesn't fix it in matplotlib (see this [thread](http://stackoverflow.com/questions/7836730/latex-on-python-alpha-and-beta-dont-work)).

In other words, even if HoloViews can accept ``'$\alpha$'``, matplotlib will still demand ``r'$\alpha$'`` or ``'$\\alpha$'``. Things I could do:

1. Always raise an exception for control codes. Problem this would disable  ``'\n$foo^3$'`` (i.e newlines and tabs).
2. Raise an exception - at least the useless bell control sequences i.e '\a' and '\b'.
3. Leave it as it is now.

Option 2. seems fairly sensible to me - if you are using LaTeX with either '\alpha' or '\beta', matplotlib will choke on these codes anyway.

**Comment by philippjfr** [comment]
I agree with option 2, it confused the hell out of me even though I'd encountered the bug before. So raising an exception for unescaped ``\a`` and ``\b`` sounds like a good idea to me.

**Comment by jlstevens** [comment]
I've addresses this issue in the PR above. An exception will now be raised which makes it clearer what has gone wrong so the user can figure out how to fix the problem.
